### ISSUE TITLE
"RIGHT" byte is not working.

### ISSUE THREAD

**Issue by ghost** [joseph346/dots#3]
1129749557781494627379878531505966426613484209395012547374631700716632035390745437596831817920210985878226506950964978169942026455068937274179474953918671824023 dots:
`
Traceback (most recent call last):
  File "Chess.py", line 127, in <module>
    main()
  File "Chess.py", line 123, in main
    run(program)
  File "Chess.py", line 83, in run
    mem.append(0)
MemoryError
`
Can you fix this?

**Comment by emmaconnor** [comment]
It sounds like you're simply running out of memory. If you move right past the end of the memory array, the interpreter tries to resize the array to match. If you move right enough then eventually the array will grow indefinitely until you run out of memory and get a memory error. This is expected behavior.

**Comment by ghost** [comment]
Okay, but I only moved once.
Here is the code in binary
`010 010 010 ... for 72 "010"s 111 000 010 010 010... for 101 "010"s 111`

**Comment by ghost** [comment]
I examined the code, and it looks like there's a bug.
The LEFT operator is fixed, not the RIGHT.
I added a proposal to change it to the correct way.

**Comment by emmaconnor** [comment]
I see the issue. The comparison:

```
            while pointer <= len(mem):
                mem.append(0)
```

Is incorrect and will cause the program to infinitely increase the size of the array. It should be `while pointer >= len(mem)` which will cause the array to resize only until it is large enough for the current pointer value. Fixed in 7f744c046033b2e9ce4241fd44abbbc2f5307ab1.
